llarp/path/pathbuilder.cpp

```diff
@@ -205,56 +205,37 @@ namespace llarp
       return obj;
     }
 
-    bool
-    Builder::SelectHop(
-        llarp_nodedb* db,
-        const std::set<RouterID>& exclude,
-        RouterContact& cur,
-        size_t hop,
-        PathRole roles)
+    std::optional<RouterContact>
+    Builder::SelectFirstHop() const
     {
-      (void)roles;
-      size_t tries = 10;
-      if (hop == 0)
-      {
-        if (m_router->NumberOfConnectedRouters() == 0)
-        {
-          return false;
-        }
-        bool got = false;
-        m_router->ForEachPeer(
-            [&](const ILinkSession* s, bool isOutbound) {
-              if (s && s->IsEstablished() && isOutbound && !got)
-              {
-                const RouterContact rc = s->GetRemoteRC();
-#ifdef TESTNET
-                if (got || exclude.count(rc.pubkey))
-#else
-                if (got || exclude.count(rc.pubkey) || m_router->IsBootstrapNode(rc.pubkey))
+      std::optional<RouterContact> found = std::nullopt;
+      m_router->ForEachPeer(
+          [&](const ILinkSession* s, bool isOutbound) {
+            if (s && s->IsEstablished() && isOutbound && not found.has_value())
+            {
+              const RouterContact rc = s->GetRemoteRC();
+#ifndef TESTNET
+              if (m_router->IsBootstrapNode(rc.pubkey))
+                return;
 #endif
-                  return;
-                cur = rc;
-                got = true;
-              }
-            },
-            true);
-        return got;
-      }
+              found = rc;
+            }
+          },
+          true);
+      return found;
+    }
 
-      do
+    std::optional<std::vector<RouterContact>>
+    Builder::GetHopsForBuild()
+    {
+      if (const auto maybe = m_router->nodedb()->GetIf([r = m_router](const auto& rc) -> bool {
```

Comment:
put the `maybe` assignment outside the if please, or remove the `; maybe.has_value()` as it is redundant in this context.

Reply:
also, is `GetIf` the correct function here?  Seems like we want a random/arbitrary RC, but `GetIf` sounds like it should give us an RC we specify *if* it meets the condition we specify.